py/server/deephaven_internal/auto_completer/_signature_help.py

```diff
@@ -0,0 +1,245 @@
+    """
+
+    param_docs = []
+    for i in range(len(signature.params)):
+        if signature.params[i].to_string().strip() in IGNORE_PARAM_NAMES:
+            continue
+
+        param = params[i]
+        description = f"##### **{param['name']}**"
+        if param['type'] is not None:
+            description += f" : *{param['type']}*"
+        description += "\n\n"
+        if param['description'] is not None:
+            description += f"{param['description']}\n\n"
+        description += "---"
+
+        param_docs.append([signature.params[i].to_string().strip(), description])
+
+    return param_docs
+
+
+def _get_signature_help(signature: Signature) -> list[Any]:
+    """ Gets the result of a signature to be used by `do_signature_help`
+
+    If no docstring information is parsed, then the signature will be displayed in Markdown but with plaintext style
+    whitespace. Any cached docstring must have some docstring information.
+
+    Returns:
+        A list that contains [signature name, docstring, param docstrings, index]
+    """
+
+    docstring = signature.docstring(raw=True)
+    cache_key = _hash(signature)
+
+    # Check cache
+    if cache_key in result_cache:
+        result = result_cache[cache_key]
+        return [
+            _generate_display_sig(signature),
+            result["description"],
+            result["param_docs"],
+            signature.index if signature.index is not None else -1,
+        ]
+
+    # Parse the docstring to extract information
+    docs = parse(docstring)
+    # Nothing parsed, revert to plaintext
+    # Based on code, the meta attribute seems to be a list of parsed items. Then, the parse function returns the
+    #   style with the most amount of items in meta. If there are no items, that should be mean nothing was parsed.
+    # https://github.com/rr-/docstring_parser/blob/4951137875e79b438d52a18ac971ec0c28ef269c/docstring_parser/parser.py#L46
+    if len(docs.meta) == 0:
+        return [
+            signature.to_string(),
+            # Since signature is a markdown, replace whitespace in a way to preserve how it originally looks
+            signature.docstring(raw=True).replace(" ", "&nbsp;").replace("\n", "  \n"),
+            [[param.to_string().strip(), ""] for param in signature.params],
+            signature.index if signature.index is not None else -1,
+        ]
+
+    # Get params in this scope because it'll be used multiple times
+    params = _get_params(signature, docs)
+    description = _generate_description_markdown(docs, params)
+    param_docs = _generate_param_markdowns(signature, params)
+    result_cache[cache_key] = {
+        "description": description,
+        "param_docs": param_docs,
+    }
+
+    return [
+        _generate_display_sig(signature),
+        description,
+        param_docs,
+        signature.index if signature.index is not None else -1,
+    ]
```

Comment:
Should this be a class rather than a list in order to make it easier to work with?

Reply:
This is returned to Java, and was the format being done prior to this change. Was not looking to refactor it.